generator/wrapper_gen.py

```diff
@@ -245,6 +251,240 @@ def daal_generate_shuffled_indices(idx, random_state):
     c_generate_shuffled_indices(data_or_file(<PyObject*>idx), data_or_file(<PyObject*>random_state))
 
 
+def get_gbt_model_from_lgbm(model: Any) -> Any:
+    class Node:
+        def __init__(self, tree: Dict[str, Any], parent_id: int, position: int):
+            self.tree = tree
+            self.parent_id = parent_id
+            self.position = position
+
+    lgb_model = model.dump_model()
+
+    n_features = lgb_model["max_feature_idx"] + 1
+    n_iterations = len(lgb_model["tree_info"]) / lgb_model["num_tree_per_iteration"]
+    n_classes = lgb_model["num_tree_per_iteration"]
+
+    is_regression = False
+    objective_fun = lgb_model["objective"]
+    print("OBJ:   " + objective_fun)
+    if n_classes > 2:
+        if "multiclass" in objective_fun:
+            print("Found multiclass classification")
+        else:
+            raise TypeError(
+                "multiclass (softmax) objective is only supported for multiclass classification")
+    elif "binary" in objective_fun:  # nClasses == 1
+        print("Found binary classification")
+        n_classes = 2
+    else:
+        print("Found regression")
+        is_regression = True
+
+    if is_regression:
+        mb = gbt_reg_model_builder(n_features=n_features, n_iterations=n_iterations)
+    else:
+        mb = gbt_clf_model_builder(
+            n_features=n_features, n_iterations=n_iterations, n_classes=n_classes)
+
+    class_label = 0
+    iterations_counter = 0
+    for tree in lgb_model["tree_info"]:
+        if is_regression:
+            tree_id = mb.create_tree(tree["num_leaves"]*2-1)
+        else:
+            tree_id = mb.create_tree(n_nodes=tree["num_leaves"]*2-1, class_label=class_label)
```

Comment:
seems `n_nodes=tree["num_leaves"]*2-1` is applicable only for complete binary tree. Do you have some evidence that trees in lgbm model are always compleat?
Seems it would be better to calculate number of nodes by initial look at model.

Reply:
The fact that the tests are working indirectly proves, that we always have a complete binary tree. When I tried to call `create_tree()` with similar arguments in xgboost convertor, where trees are often incomplete, I always got a memory error while building model in daal. And I never met such errors in LightGBM convertor, so I think, this code might be considered as applicable.